### ISSUE TITLE
crash in createOptionalType()

### ISSUE THREAD

**Issue by judej** [microsoft/pyright#8524]
Several crash reports with the following stack trace:

```
TypeError: Cannot read properties of undefined (reading 'type')
at type  /analyzer/typeEvaluator.ts#L14959'>pyright-internal/src/analyzer/typeEvaluator.ts:14959:39 
at createOptionalType  /analyzer/typeEvaluator.ts#L20012'>pyright-internal/src/analyzer/typeEvaluator.ts:20012:35 
     at None ......src/languageService/completionProvider.ts:499:33 
at callback  /analyzer/typeUtils.ts#L799'>pyright-internal/src/analyzer/typeUtils.ts:799:8 
at expandedType  /analyzer/typeEvaluator.ts#L3954'>pyright-internal/src/analyzer/typeEvaluator.ts:3954:16 
at expandSubtype  /analyzer/typeEvaluator.ts#L4011'>pyright-internal/src/analyzer/typeEvaluator.ts:4011:12 
at mapSubtypesExpandTypeVars  /analyzer/typeEvaluator.ts#L7017'>pyright-internal/src/analyzer/typeEvaluator.ts:7017:21 
    at None ....../src/intelliCode/tokens/lookbackTokenGenerator.ts:24:12 
at node  /analyzer/typeEvaluator.ts#L1068'>pyright-internal/src/analyzer/typeEvaluator.ts:1068:49 
at getTypeOfExpression  /analyzer/typeEvaluator.ts#L1813'>pyright-internal/src/analyzer/typeEvaluator.ts:1813:31 
at getTypeOfAnnotation  /analyzer/typeEvaluator.ts#L19434'>pyright-internal/src/analyzer/typeEvaluator.ts:19434:12 
at callback  /analyzer/typeEvaluator.ts#L19888'>pyright-internal/src/analyzer/typeEvaluator.ts:19888:8 
at evaluateTypeForSubnode  ../analyzer/typeEvaluator.ts#L856'>pyright-internal/src../analyzer/typeEvaluator.ts:856:19 
at callback  /common/timing.ts#L40'>pyright-internal/src/common/timing.ts:40:19 
at getType  ../analyzer/checker.ts#L1236'>pyright-internal/src../analyzer/checker.ts:1236:24 
at visitIndex  ../analyzer/parseTreeWalker.ts#L439'>pyright-internal/src../analyzer/parseTreeWalker.ts:439:28 
at visit  ../analyzer/parseTreeWalker.ts#L933'>pyright-internal/src../analyzer/parseTreeWalker.ts:933:20 
at visitNode  ../analyzer/parseTreeWalker.ts#L915'>pyright-internal/src../analyzer/parseTreeWalker.ts:915:36 
at walk  ../analyzer/checker.ts#L282'>pyright-internal/src../analyzer/checker.ts:282:18 
    at walk  ../analyzer/checker.ts#L585'>pyright-internal/src../analyzer/checker.ts:585:21 
at func (<anonymous>)
at forEach  ../analyzer/checker.ts#L579'>pyright-internal/src../analyzer/checker.ts:579:22 
at visitFunction  ../analyzer/parseTreeWalker.ts#L451'>pyright-internal/src../analyzer/parseTreeWalker.ts:451:28 
at visit  ../analyzer/parseTreeWalker.ts#L933'>pyright-internal/src../analyzer/parseTreeWalker.ts:933:20 
at visitNode  ../analyzer/parseTreeWalker.ts#L915'>pyright-internal/src../analyzer/parseTreeWalker.ts:915:36 
at walk  ../analyzer/checker.ts#L282'>pyright-internal/src../analyzer/checker.ts:282:18 
at walk  ../analyzer/checker.ts#L2837'>pyright-internal/src../analyzer/checker.ts:2837:17 
at _walkStatementsAndReportUnreachable  ../analyzer/checker.ts#L291'>pyright-internal/src../analyzer/checker.ts:291:13 
at visitSuite  ../analyzer/parseTreeWalker.ts#L544'>pyright-internal/src../analyzer/parseTreeWalker.ts:544:28 

```

**Comment by erictraut** [comment]
Thanks for reporting. This occurs when the user types `Optional[]`. This is a syntax error, so it won't appear in real code, but it can appear momentarily when typing the code due to auto-insertion of the right square brace.

This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.374.
